### DISCUSSION TITLE
Ability to toggle visible to true or false on Disclosures

### DISCUSSION THREAD

**Discussion by bvsillorequez** [twilio-labs/paste#1041] [Q&A] [answered]
I have a sub-menu that needs to be toggled visible when an item is selected in that sub-menu.  Currently when visible is hardcoded to true or false it shows the correct state of the Disclosure.  
![Screen Shot 2020-12-14 at 12 48 57 PM](https://user-images.githubusercontent.com/60529554/102128215-6331d900-3e02-11eb-844f-9eb347e7d668.png)
![Screen Shot 2020-12-14 at 12 47 48 PM](https://user-images.githubusercontent.com/60529554/102128218-63ca6f80-3e02-11eb-9ff2-957cb04e7d1e.png)

When I set visible to change between true or false to display what I need to display that submenu if something in that submenu is selected but when set to true like the photo below, the menu doesn't open.  Sidenote:  Manage is red because it is set in a Disclosure heading and I was checking to see if our logic was correct and would change colors if an item in the menu is selected.

![Screen Shot 2020-12-14 at 12 52 21 PM](https://user-images.githubusercontent.com/60529554/102128718-21556280-3e03-11eb-9696-e58454e5c40f.png)

Is there a way to change or modify visible so that when an item in that menu is selected the Disclosure opens up?

**Comment by richbachman** [comment, marked as answer]
Hi @bvsillorequez

One solution might be to use the `useDisclosureState` hook and `state` prop: https://paste.twilio.design/components/disclosure#using-state-hooks

That allows you to tie into the hook happening within the Disclosure. I've set up a very simple example that might help you out here: https://codesandbox.io/s/disclosure-state-ebl15?file=/src/index.tsx

Let us know if that helps.

**Comment by bvsillorequez** [comment]
@richbachman Thank you. That helped.  I was trying to avoid it due to the warning about using state hooks but it solved the problem I had.
